src/utils/presets/withdrawInflationAndPegin.js

```diff
@@ -0,0 +1,115 @@
+import { ethers } from 'ethers'
+
+import { ethPNTContract, pNetworkV2Vault, prepareInflationData, prepareInflationProposal } from './utils'
+import settings from '../../settings'
+
+const withdrawInflationAndPegin = ({ presetParams, setPresetParams }) => ({
+  id: 'withdrawInflationAndPegin',
+  name: 'Withdraw Inflation and Pegin',
+  description: 'Withdraw requested inflated ethPNT amount to the treasury and Pegin to an account on a different chain',
+  args: [
+    {
+      id: 'input-amount',
+      name: 'amount',
+      component: 'Input',
+      props: {
+        type: 'number',
+        style: {
+          fontSize: 15
+        },
+        placeholder: 'Amount ...',
+        value: presetParams[0] || '',
+        onChange: (_e) =>
+          setPresetParams({
+            ...presetParams,
+            0: _e.target.value
+          })
+      }
+    },
+    {
+      id: 'select-chain',
+      name: 'destinationChain',
+      component: 'ChainSelection',
+      props: {
+        onSelect: (_chainId) =>
+          setPresetParams({
+            ...presetParams,
+            1: _chainId
+          })
+      }
+    },
+    {
+      id: 'input-receiver-address',
+      name: 'receiverAddress',
+      component: 'Input',
+      props: {
+        style: {
+          fontSize: 15
+        },
+        placeholder: 'Receiver address ...',
+        value: presetParams[2] || '',
+        onChange: (_e) =>
+          setPresetParams({
+            ...presetParams,
+            2: _e.target.value
+          })
+      }
+    },
+    {
+      id: 'input-user-data',
+      name: 'userData',
+      component: 'Input',
+      props: {
+        style: {
+          fontSize: 15
+        },
+        placeholder: 'Optional User Data ...',
+        value: presetParams[3] || '',
+        onChange: (_e) =>
+          setPresetParams({
+            ...presetParams,
+            3: _e.target.value
+          })
+      }
+    }
+  ],
+  prepare: async () => {
+    if (Object.values(presetParams).length < 2) return null
+    const amount = presetParams[0]
+    const destinationChainId = !presetParams[1] ? settings.chains[0].chainId : presetParams[1]
+    const destinationAddress = presetParams[2]
+    const userData = presetParams[3] ? presetParams[3] : '0x'
+
+    if (!ethers.utils.isAddress(destinationAddress)) throw new Error('Inserted destination address is not valid')
+
+    const inflationData = prepareInflationData(amount)
+    const inflationProposal = await prepareInflationProposal(
+      inflationData.ethPNTAddress,
+      settings.contracts.dandelionVoting,
+      inflationData.rawAmount
+    )
+
+    const approve = {
```

Comment:
Are we skipping a step here?
Isn't the treasury the only admissible receiver for the inflation? Thus, there should be a transfer from the treasury to dandelion before approving.

Reply:
That's included into the `prepareInflationData()` fn

Reply:
cc @gitmp01

Reply:
Checked and LGTM.